x/leverage/keeper/inspector.go

```diff
@@ -0,0 +1,363 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sort"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/umee-network/umee/v5/x/leverage/types"
+)
+
+// Separated from grpc_query.go
+func (q Querier) Inspect(
+	goCtx context.Context,
+	req *types.QueryInspect,
+) (*types.QueryInspectResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// This query is also disabled by default as a safety measure. Enable with liquidator queries.
+	if !q.Keeper.liquidatorQueryEnabled {
+		return nil, types.ErrNotLiquidatorNode
+	}
+
+	k, ctx := q.Keeper, sdk.UnwrapSDKContext(goCtx)
+	var filters []inspectorFilter
+	var sorting inspectorSort
+
+	// The "all" symbol denom is converted to empty symbol denom
+	if strings.EqualFold(req.Symbol, "all") {
+		req.Symbol = ""
+	}
+	modeMin := sdk.MustNewDecFromStr(fmt.Sprintf("%f", req.ModeMin))
+	sortMin := sdk.MustNewDecFromStr(fmt.Sprintf("%f", req.SortMin))
+	specific := req.Symbol != ""
+
+	switch strings.ToLower(req.Mode) {
+	case "borrowed":
+		filters = append(filters, withMinBorrowedValue(modeMin, specific))
+	case "collateral":
+		filters = append(filters, withMinCollateralValue(modeMin, specific))
+	case "danger":
+		filters = append(filters, withMinDanger(modeMin))
+	case "ltv":
+		filters = append(filters, withMinLTV(modeMin))
+	case "zeroes":
+		filters = append(filters, withZeroes())
+	default:
+		return &types.QueryInspectResponse{}, status.Error(codes.InvalidArgument, "unknown inspector mode: "+req.Mode)
+	}
+	switch strings.ToLower(req.Sort) {
+	case "borrowed":
+		sorting = moreBorrowed(specific)
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinBorrowedValue(sortMin, specific))
+		}
+	case "collateral":
+		sorting = moreCollateral(specific)
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinCollateralValue(sortMin, specific))
+		}
+	case "danger":
+		sorting = moreDanger()
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinDanger(sortMin))
+		}
+	case "ltv":
+		sorting = moreLTV()
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinLTV(sortMin))
+		}
+	default:
+		// if no sort mode is specified, return all borrowers sorted by total collateral value
+		sorting = moreCollateral(false)
```

Comment:
why we don't add `sortMin` filter here?
I think the `if sortMin.isPositive()` .... should be outside of the switch, then being repeated.

Reply:
`if sortMin.isPositive()` does a different thing depending on which case it's in, so it can't be moved outside

In the default (no sorting specified) case, we currently ignore the minimum value to filter from sorted collateral.

Reply:
simplified